dask/array/reductions.py

```diff
@@ -225,8 +245,8 @@ def nannumel(x, **kwargs):
 def mean_chunk(x, sum=chunk.sum, numel=numel, dtype='f8', **kwargs):
     n = numel(x, dtype=dtype, **kwargs)
     total = sum(x, dtype=dtype, **kwargs)
-    result = np.empty(shape=n.shape,
-                      dtype=[('total', total.dtype), ('n', n.dtype)])
+    result = empty_of_type(n, shape=n.shape,
+                           dtype=[('total', total.dtype), ('n', n.dtype)])
```

Comment:
I have a slight preference to explicitly perform the dispatch above and then use the `empty` function below.  This is just because `empty` happens to be a well known term to numpy users.